### PR TITLE
chore: add / for forward compatibility of the lockfile

### PR DESCRIPTION
**Pull request by ruben-arts**
This should re-add the backslash to the channel.

This is to be forward compatible with the previous situation.

### CODE REVIEW

File: crates/rattler_conda_types/src/channel/mod.rs [hunk 1]
```diff
@@ -94,7 +97,7 @@ impl NamedChannelOrUrl {
                 }
                 base_url
```
Comment (baszalmstra) [inline crates/rattler_conda_types/src/channel/mod.rs:98]: This is not guaranteed to have a trailing slash now.

Comment (baszalmstra) [reply crates/rattler_conda_types/src/channel/mod.rs:98]: Maybe you can also add a comment to the function that the returned url will always have a trailing slash?

Comment (baszalmstra) [reply crates/rattler_conda_types/src/channel/mod.rs:98]: And also add some tests to verify that base_url does always have a trailing slash but as_str doesnt.

Comment (ruben-arts) [reply crates/rattler_conda_types/src/channel/mod.rs]: I think i've added what you requested in this comment but I'm not sure what you mean with the `as_str`. The canonical name is tested in:
https://github.com/mamba-org/rattler/blob/68c4254de11293639a5800b6f05295405b2d5eff/crates/rattler_conda_types/src/channel/mod.rs#L654-L673

File: crates/rattler_conda_types/src/channel/mod.rs [hunk 2]
```diff
@@ -85,16 +88,17 @@ impl NamedChannelOrUrl {
...
+    /// This method ensures that the base url always ends with a `/`.
     pub fn into_base_url(self, config: &ChannelConfig) -> Url {
         match self {
             NamedChannelOrUrl::Name(name) => {
                 let mut base_url = config.channel_alias.clone();
                 if let Ok(mut segments) = base_url.path_segments_mut() {
                     segments.push(&name);
                 }
-                base_url
+                add_trailing_slash(&base_url).into_owned()
```
Comment (baszalmstra) [inline crates/rattler_conda_types/src/channel/mod.rs:99]: This line is a duplicate of the line below. Please return something from the match and then apply removing the trailing slash on the return value.

File: crates/rattler_conda_types/src/channel/mod.rs [hunk 3]
```diff
@@ -667,4 +671,41 @@ mod tests {
...
+        let channel = Channel::from_str("conda-forge", &channel_config).unwrap();
+        let channel_with_backslash =
+            Channel::from_str("https://conda.anaconda.org/conda-forge/", &channel_config).unwrap();
+        assert_eq!(channel.base_url(), channel_with_backslash.base_url());
+
+        // Channel with backslash should have backslash
+        let channel_with_backslash = Channel::from_str("conda-forge/", &channel_config).unwrap();
+        let channel =
+            Channel::from_str("https://conda.anaconda.org/conda-forge", &channel_config).unwrap();
+        assert_eq!(channel.base_url(), channel_with_backslash.base_url());
```
Comment (baszalmstra) [inline crates/rattler_conda_types/src/channel/mod.rs:692]: Here you are basically checking if for some input when you parse it to a channel it contains a backslash. But there is a lot of duplication. Maybe write a for loop over some input values and in the for loop check if the base_url contains a backslash. You can then also add more input values. Try with a url that already contains a backslash etc.

File: crates/rattler_conda_types/src/channel/mod.rs [hunk 4]
```diff
@@ -667,4 +671,41 @@ mod tests {
...
+
+        // The named channel to channel should have backslash
+        let channel = named_channel.into_channel(&channel_config);
+        let channel_with_backslash =
+            Channel::from_str("https://conda.anaconda.org/conda-forge/", &channel_config).unwrap();
+        assert_eq!(channel.base_url(), channel_with_backslash.base_url());
+        assert_eq!(
+            channel.base_url().as_str(),
+            "https://conda.anaconda.org/conda-forge/"
+        );
```
Comment (baszalmstra) [inline crates/rattler_conda_types/src/channel/mod.rs]: Some here!

File: crates/rattler_conda_types/src/channel/mod.rs (hunk 3 above)
Comment (ruben-arts) [reply crates/rattler_conda_types/src/channel/mod.rs]: :+1:

File: crates/rattler_conda_types/src/channel/mod.rs (hunk 4 above)
Comment (ruben-arts) [reply crates/rattler_conda_types/src/channel/mod.rs]: :+1:

File: crates/rattler_conda_types/src/channel/mod.rs (hunk 2 above)
Comment (ruben-arts) [reply crates/rattler_conda_types/src/channel/mod.rs]: Cleaned it up

**Review by baszalmstra — APPROVED** [review]

**Pull request merged**